app.py

```diff
@@ -1,2 +1,46 @@
-from flask import Flask, request
+from flask import Flask, request, jsonify
 import json
+from flask.views import MethodView
+from werkzeug.routing import ValidationError
+
+app = Flask(__name__)
+with open("restaurants.json", "r") as read_file:
+    data = json.load(read_file)
+
+restaurants = data['restaurants']
+
+
+class RestaurantEndpoint(MethodView):
+    def get(self):
+        return jsonify(restaurants), 200
+
+    def post(self):
+        content = request.json
+        if content is None:
+            raise ValidationError
+        restaurants.append(content)
+        return jsonify({'restaurants': restaurants}), 201
+
+
+class RestaurantItemEndpoint(MethodView):
+    def get(self, restaurant_id):
+        return jsonify(restaurants[restaurant_id]), 200
+
+    def put(self, restaurant_id):
+        content = request.json
+        answer = 200
+        if content is not None:
+            if len(restaurants) <= restaurant_id:
+                answer = 201
+                restaurants.append(content)
+            else:
+                restaurants[restaurant_id] = content
+        return jsonify(restaurants[restaurant_id]), answer
```

Comment:
`restaraunt_id` does not have to follow order of items in the list, `restaraunt_id` is `item.id`.
 you should scan the list and find an item where `item.id == restaurant_id`

Reply:
fixed